### PR TITLE
`uint256` hints

### LINKED ISSUE #189: Add `uint256` module hints
- `UINT256_ADD`
- `UINT256_ADD_LOW`
- `SPLIT_64`
- `UINT256_SQRT`
-  `UINT256_SQRT_FELT`
- `UINT256_SIGNED_NN`
- `UINT256_UNSIGNED_DIV_REM`
- `UINT256_EXPANDED_UNSIGNED_DIV_REM`
- `UINT256_MUL_DIV_MOD`
- `UINT256_SUB`

### PR DESCRIPTION
**Pull request by toni-calvin**

### CODE REVIEW

File: pkg/hints/uint256_hints.go [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+
+		sumHigh := aHigh.Add(bHigh.Add(carryLow))
+		var carryHigh Felt
+		switch sumHigh.Cmp(shift) {
+		case -1:
+			carryHigh = FeltZero()
+		default:
+			carryHigh = FeltOne()
+		}
+		ids.Insert("carry_high", NewMaybeRelocatableFelt(carryHigh), vm)
```
Comment (fmoletta) [inline pkg/hints/uint256_hints.go]: Unchecked insert

File: pkg/hints/hint_utils/ids_manager.go [hunk 2]
```diff
@@ -76,6 +76,18 @@ func (ids *IdsManager) GetFelt(name string, vm *VirtualMachine) (lambdaworks.Fel
...
 
+func (ids *IdsManager) GetUint256(name string, vm *VirtualMachine) (Uint256, error) {
+	low, err := ids.GetStructFieldFelt(name, 0, vm)
+	if err != nil {
+		return Uint256{}, err
+	}
+	high, err := ids.GetStructFieldFelt(name, 1, vm)
+	if err != nil {
+		return Uint256{}, err
+	}
```
Comment (fmoletta) [inline pkg/hints/hint_utils/ids_manager.go]: Its better to use `GetAddr` and then `Memory.Get` on addr  and addr + 1. This way we don't have to compute the base address of the ids twice. For an example, look at the functions in hint_utils/bigint_utils.go

File: pkg/hints/hint_utils/ids_manager.go [hunk 3]
```diff
@@ -217,6 +229,21 @@ func (ids *IdsManager) InsertStructField(name string, field_off uint, value *May
...
 
+// Inserts Uint256 value into an ids field (given the identifier is a Uint256)
+func (ids *IdsManager) InsertUint256(name string, val Uint256, vm *VirtualMachine) error {
+	err := ids.InsertStructField(name, 0, NewMaybeRelocatableFelt(val.Low), vm)
+	if err != nil {
+		return err
+	}
+	return ids.InsertStructField(name, 1, NewMaybeRelocatableFelt(val.High), vm)
+
+}
```
Comment (fmoletta) [inline pkg/hints/hint_utils/ids_manager.go]: Similar comment to GetUint256, use `GetAddr` to get the base address and then use `Memory.Insert` on addr and addr+1

File: pkg/hints/hint_utils/ids_manager.go [hunk 4]
```diff
@@ -217,6 +229,21 @@ func (ids *IdsManager) InsertStructField(name string, field_off uint, value *May
...
+
+// Inserts value into ap address
+func (ids *IdsManager) InsertValueIntoAP(vm *VirtualMachine, value MaybeRelocatable) error {
+	return vm.Segments.Memory.Insert(vm.RunContext.Ap, &value)
+}
+
```
Comment (fmoletta) [inline pkg/hints/hint_utils/ids_manager.go]: This shouldn't be an` IdsManager` method

File: cairo_programs/uint256_root.cairo [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+    Uint256,
+    uint256_sqrt,
+)
+from starkware.cairo.common.cairo_builtins import BitwiseBuiltin
+
+func main{range_check_ptr: felt, bitwise_ptr: BitwiseBuiltin*}() {
+    let n = Uint256(0, 157560248172239344387757911110183813120);
+    let res = uint256_sqrt(n);
+    return ();
+}
```
Comment (fmoletta) [inline cairo_programs/uint256_root.cairo]: Please add newline

**Review by fmoletta — APPROVED** [review]

File: cairo_programs/uint256_root.cairo (hunk 5 above)
Comment (toni-calvin) [inline cairo_programs/uint256_root.cairo]:
```suggestion
}

```

**Review by pefontana — APPROVED** [review]

**Pull request merged**